doc/decisions/error_handling.md

```diff
@@ -0,0 +1,39 @@
+# Error Handling
+
+## Problem
+
+Currently, a lot of functions return the error value, regardless of error. These poses some problems, for instance authors of bindings, where it is hard to tell which error exactly occurred. A big improvement to this would be returning different error codes for different types of errors, in order to be able to provide better information in cases of errors. Some functions also make use of an `errorKey`.
```

Comment:
Be more concrete. Afaik the only problems are:

- name modifications which can be either invalid name or locking the key name
- getting values of (non-)binary keys, see also #4194